src/realm/group_shared.cpp

```diff
@@ -1197,6 +1198,11 @@ void SharedGroup::do_open(const std::string& path, bool no_create_file, bool is_
     // Upgrade file format and/or history schema
     try {
         using gf = _impl::GroupFriend;
+        if (stored_hist_schema_version == -1) {
+            // current_hist_schema_version has not been read. Read it now
+            ReadTransaction rt(*this);
+            stored_hist_schema_version = gf::get_history_schema_version(m_group.m_alloc, m_read_lock.m_top_ref);
+        }
```

Comment:
A good benchmark would be something like this:

    SharedGroup sg{path};
    for (...) {
        SharedGroup{path};
    }

Some real world uses cases would look essentially like this I suppose.